sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareFileSeekableByteChannelWriteOptions.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.storage.file.share.options;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.file.share.models.FileLastWrittenMode;
+import com.azure.storage.file.share.models.ShareRequestConditions;
+
+import java.util.Objects;
+
+/**
+ * Options for obtaining a {@link java.nio.channels.SeekableByteChannel} backed by an Azure Storage Share File.
+ */
+public final class ShareFileSeekableByteChannelWriteOptions {
+    private static final ClientLogger LOGGER = new ClientLogger(ShareFileSeekableByteChannelWriteOptions.class);
+
+    /**
+     * Mode to open the channel for writing.
+     */
+    public enum WriteMode {
+        /**
+         * Opens channel to an existing file for writing.
+         */
+        WRITE,
+
+        /**
+         * Creates a new file for writing and opens the channel. If the file already exists, it will be overwritten.
+         * Requires a value be set with {@link ShareFileSeekableByteChannelWriteOptions#setFileSize(Long)}.
+         */
+        OVERWRITE
+    }
+
+    private final WriteMode channelMode;
+    private Long fileSize;
+    private ShareRequestConditions requestConditions;
+    private FileLastWrittenMode fileLastWrittenMode;
+
+    /**
+     * Options constructor.
+     * @param mode What usage mode to open the channel in.
+     */
+    public ShareFileSeekableByteChannelWriteOptions(WriteMode mode) {
+        channelMode = Objects.requireNonNull(mode, "'mode' cannot be null.");
```

Comment:
Do you not want a constructor to include requestConditions and fileLastWrittenMode?

Reply:
We have fluent setters for that. The pattern in our java library has been to use required parameters in the constructor and optional ones through fluent setters, since otherwise object construction can get rather messy due to java not having a concept of optional parameters. We don't want a future with `new ShareFileSeekableByteChannelWriteOptions(mode, null, null, 0, null, false, null, null)`.

Reply:
Ah, ok, makes sense, thanks. And wow, I wasn't aware that Java didn't have Optimal params!